task.sql

```diff
@@ -2,10 +1,15 @@
 USE ShopDB; 
 
--- Some data should be created outside the transaction (here)
-
--- Start the transaction 
 START TRANSACTION; 
 
--- And some data should be created inside the transaction 
+INSERT INTO Orders (CustomerID, Date) 
+VALUES (1, '2023-01-01');
+
+INSERT INTO OrderItems (OrderID, ProductID, Count)
+VALUES (LAST_INSERT_ID(), 1, 1);
+
+UPDATE Products 
+SET WarehouseAmount = WarehouseAmount - 1 
+WHERE ID = 1;
```

Comment:
Ensure that the `Products` table has a product with `ID = 1` and that the `WarehouseAmount` is sufficient to be decremented by 1. If not, this could lead to negative stock values or errors.